edb/edgeql/compiler/viewgen.py

```diff
@@ -244,6 +250,100 @@ def _process_view(
             )
         )
 
+    explicit_ptr_names = {
+        desc.ptr_name for desc in shape_desc if not desc.is_linkprop
+    }
+
+    explicit_lprop_names = {
+        desc.ptr_name for desc in shape_desc if desc.is_linkprop
+    }
+
+    # Now look for any splats and expand them.
+    for shape_el in elements:
+        if not isinstance(shape_el.expr.steps[0], qlast.Splat):
+            continue
+
+        if exprtype is not s_types.ExprType.Select:
+            raise errors.QueryError(
+                "unexpected splat operator in non-SELECT shape",
+                context=shape_el.expr.context,
+            )
+
+        if ctx.env.options.func_params is not None:
```

Comment:
We might want to make sure to disable it in other schema stuff, like constraints and indexes? (Though I think trying to do that will fail for other reasons, so)

Reply:
We don't allow anything complex in those contexts already.